sympy/integrals/transforms.py

```diff
@@ -1008,9 +1008,34 @@ def repl(ex, *args):
 def _laplace_transform(f, t, s_, simplify=True):
     """ The backend function for Laplace transforms. """
     from sympy import (re, Max, exp, pi, Min, periodic_argument as arg_,
-                       arg, cos, Wild, symbols, polar_lift)
+                       arg, cos, Wild, symbols, polar_lift, limit,
+                       DiracDelta)
     s = Dummy('s')
-    F = integrate(exp(-s*t) * f, (t, 0, oo))
+    zero_minus = symbols('zero_minus', real=True)
+    # Expressions with DiracDelta need to be computed with a directional
+    # limit, but this will fail on some sums, so we first use the
+    # linearity of the Laplace transform to split the integral into a sum
+    # of integrals.
+    if f.has(DiracDelta):
+        f = f.expand(deep=True, modulus=None, power_base=False,
+                     power_exp=False, mul=True, log=False, multinomial=True,
+                     basic=True, complex=False, func=False, trig=False)
+        if f.is_Add:
+            _all_parts = []
+            for _part in f.args:
+                if _part.has(DiracDelta(t)):
```

Comment:
How do you plan to handle expressions that contain something like `DiracDelta(2*t)`?

Reply:
Nice catch! I corrected that incorrectly in commit https://github.com/sympy/sympy/pull/21469/commits/7ba0a99e76c8f4659a3dc8bb457be0f78e743ccd , where I removed `(t)` only in one place instead of two places. 

The really curious thing is that it already worked like this. I will still change it with the next commit ;)